usage/camp/src/main/java/io/brooklyn/camp/brooklyn/spi/dsl/methods/DslComponent.java

```diff
@@ -122,14 +124,15 @@ public Entity call() throws Exception {
                 default:
                     throw new IllegalStateException("Unexpected scope "+scope);
             }
-            
-            Optional<Entity> result = Iterables.tryFind(entitiesToSearch, EntityPredicates.configEqualTo(BrooklynCampConstants.PLAN_ID, componentId));
+            Predicate<Entity> configEqualTo = EntityPredicates.configEqualTo(BrooklynCampConstants.PLAN_ID, componentId);
+            Predicate<Entity> classNameEquals = EntityPredicates.classNameEquals(componentId);
+            Optional<Entity> result = Iterables.tryFind(entitiesToSearch, Predicates.or(configEqualTo, classNameEquals));
```

Comment:
If the string matches both an id and a class name, then would be good to guarantee it will first match on id rather than it being dependent on the order that it walks through the entities. I'd therefore go for the more verbose:

    Optional<Entity> result = Iterables.tryFind(entitiesToSearch, configEqualTo);
    if (!result.isPresent()) result = Iterables.tryFind(entitiesToSearch, classNameEquals);